client/modules/IDE/components/PreviewFrame.js

```diff
@@ -127,12 +127,18 @@ class PreviewFrame extends React.Component {
       htmlFile = htmlFile.replace(fileRegex, `<style>\n${cssFile.content}\n</style>`);
     });
 
-    // const htmlHead = htmlFile.match(/(?:<head.*?>)([\s\S]*?)(?:<\/head>)/gmi);
-    // const headRegex = new RegExp('head', 'i');
-    // let htmlHeadContents = htmlHead[0].split(headRegex)[1];
-    // htmlHeadContents = htmlHeadContents.slice(1, htmlHeadContents.length - 2);
-    // htmlHeadContents += '<link rel="stylesheet" type="text/css" href="/preview-styles.css" />\n';
-    // htmlFile = htmlFile.replace(/(?:<head.*?>)([\s\S]*?)(?:<\/head>)/gmi, `<head>\n${htmlHeadContents}\n</head>`);
+    if (this.props.textOutput || this.props.isTextOutputPlaying) {
+      const htmlHead = htmlFile.match(/(?:<head.*?>)([\s\S]*?)(?:<\/head>)/gmi);
+      const headRegex = new RegExp('head', 'i');
+      let htmlHeadContents = htmlHead[0].split(headRegex)[1];
+      htmlHeadContents = htmlHeadContents.slice(1, htmlHeadContents.length - 2);
+      htmlHeadContents += '<script src="/loadData.js"></script>\n';
+      htmlHeadContents += '<script src="/interceptor-functions.js"></script>\n';
+      htmlHeadContents += '<script src="/intercept-p5.js"></script>\n';
+      htmlHeadContents += '<script type="text/javascript" src="http://chir.ag/projects/ntc/ntc.js"></script>';
```

Comment:
We probably want to serve this file locally as well, even though I know the person said you can link directly to this file. Just in case their site is ever down or whatever.